microsoft-edge/devtools-guide-chromium/memory-problems/memory-101.md

```diff
@@ -23,165 +23,146 @@ ms.date: 07/21/2023
-
 
 <!-- ====================================================================== -->
-## Objects retaining tree
+## Object sizes and distances
 
-The heap is a network of interconnected objects.  In the mathematical world, this structure is called a _graph_ or _memory graph_.  A graph is constructed from _nodes_ that are connected by _edges_.
+When working with the **Memory** tool, you will likely find yourself looking at the following columns of information:
 
-Nodes and edges in a graph are given labels as follows:
+* **Shallow Size**
+* **Retained Size**
+* **Distance**
 
-*  _Nodes_ (or _objects_) are labelled using the name of the _constructor_ function that was used to build them.
+The numbers that are shown in the **Shallow size** and **Retained size** columns are the number of bytes.
 
-*  _Edges_ are labelled using the names of _properties_.
-
-Learn [how to record a profile using the Heap Profiler](heap-snapshots.md).  In the following figure, some of the notable things in the Heap Snapshot recording in the [Memory](heap-snapshots.md) tool include **Distance**, which is the distance from the garbage-collection root.  If almost all the objects of the same type are at the same distance, and a few are at a bigger distance, that's something worth investigating.
-
-Distance from root:
+![Shallow and Retained Size](./memory-101-images/shallow-retained.png)
 
-![Distance from root](./memory-101-images/root.png)
+<!-- ------------------------------ -->
+#### Shallow size
 
+The _shallow size_ is the size of the memory that is held by the object directly.
 
-<!-- ====================================================================== -->
-## Dominators
+The shallow size of an object is usually small because JavaScript objects often just store their own descriptions only in memory. Arrays and strings are exceptions to this and can have a significant shallow size. Most JavaScript objects store their values elsewhere in the renderer memory, and only expose a small wrapper object on the JavaScript heap.
 
-Dominator objects are comprised of a tree structure, because each object has exactly one dominator.  A dominator of an object might lack direct references to an object that it dominates.  That is, dominator's tree is not a spanning tree of the graph.
+Nevertheless, even a small object can hold a large amount of memory indirectly, by preventing other objects from being disposed of by the automatic garbage collection process.
 
-In the following figure:
+<!-- ------------------------------ -->
+#### Retained size
 
-*  Node 1 dominates node 2.
-*  Node 2 dominates nodes 3, 4 and 6.
-*  Node 3 dominates node 5.
-*  Node 5 dominates node 8.
-*  Node 6 dominates node 7.
+The _retained size_ is the size of memory that is implicitly held by an object and that is freed once the object is deleted along with all the dependent objects that were made unreachable from GC roots.
```

Comment:
Two comments here:

1. I would say "would be freed" rather than "is freed", since we haven't yet freed the object (it's in the snapshot, after all)
2. Christy changed what this column means. I'm not really sure how to describe the meaning of the new value; you might want to consult with her. https://dev.azure.com/microsoft/Edge/_git/chromium.devtools-frontend/pullrequest/9283784

Reply:
@christyycchen any proposal for a revised wording here, that would match the new retained size column counting logic?

Reply:
Something around 
"is the size of memory that is implicitly held by an object and that could be freed once the object and other existing retainer is deleted along with all the dependent objects that were made unreachable from GC roots." ? 🤔
Maybe add some notes explaining that when a child node is retained by multiple nodes, the size of that child node is contributed to the parent node that has the shorted path to the GC root. Some graphs could help explaining this.